db/post_migrate/20220526135414_remove_corrupted_last_read_message_id.rb

```diff
@@ -0,0 +1,38 @@
+class RemoveCorruptedLastReadMessageId < ActiveRecord::Migration[7.0]
+  def change
```

Comment:
We shouldn't use ActiveRecord in migrations (to make sure the migration continues to work forever, regardless of future changes/removals of ActiveRecord models).

So I think this needs to be re-written in raw SQL statements 😓

Reply:
Totally untested, but I'm imagining something like this:

```ruby
# Delete memberships for deleted channels
execute <<~SQL
  DELETE FROM user_chat_channel_memberships
  LEFT JOIN chat_channels ON chat_channels.id = user_chat_channel_memberships.chat_channel_id
  WHERE chat_channels.id IS NULL
SQL

# Delete messages for deleted channels
execute <<~SQL
  DELETE FROM chat_messages
  LEFT JOIN chat_channels ON chat_channels.id = chat_messages.chat_channel_id
  WHERE chat_channels.id IS NULL
SQL

# Reset highest_channel_message_id if the message does not exist, or is part of a different channel
execute <<~SQL
  WITH highest_channel_message_id AS (
    SELECT chat_channel_id, max(chat_messages.id) as highest_id
    FROM chat_messages
    GROUP_BY chat_channel_id
  )
  UPDATE user_chat_channel_memberships uccm
  SET uccm.last_read_message_id = highest_channel_message_id.highest_id
  FROM highest_channel_message_id
  WHERE uccm.last_read_message_id IS NOT NULL
  AND uccm.last_read_message_id NOT IN (
    SELECT id FROM chat_messages WHERE chat_messages.chat_channel_id = uccm.chat_channel_id
  )
SQL
```

Reply:
Oh yes I re

> We shouldn't use ActiveRecord in migrations (to make sure the migration continues to work forever, regardless of future changes/removals of ActiveRecord models).
> 
> So I think this needs to be re-written in raw SQL statements 😓

Oh yes I remember rewriting many migrations in automation to raw sql... thanks